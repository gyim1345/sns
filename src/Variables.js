export const post = {
  like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
  tag: [('boat', 'sky', 'landscape')],
  _id: '5e8ec52d9bb4913b5a844423',
  id: 1,
  title: 'posting with id 1',
  imageUrl: 'http://localhost:3000/static/images/1.jpg',
  userName: 'gibong@gmail.com'
};

export const posts = [
  {
    like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
    tag: [('boat', 'sky', 'landscape')],
    _id: '5e8ec52d9bb4913b5a844423',
    id: 1,
    title: 'posting with id 1',
    imageUrl: 'http://localhost:3000/static/images/1.jpg',
    userName: 'gibong@gmail.com'
  }
];

export const currentUser = 'gibong@gmail.com';

export const currentUserName = 'gibong'

export const comments = [
  {
    like: [('gibong@gmail.com', 'guy@gmail.com', 'noone@gmail.com')],
    _id: '5e90119435a2d94f9e08e35a',
    id: 1,
    postLId: 1,
    title: 'title1',
    userName: 'gibong@gmail.com',
    replyToCommentId: null
  }
];

export const info = {
  userURL: 'http://localhost:3000/static/images/user1.png',
  name: 'gibong@gmail.com',
  postCount: 16,
  followerNumber: 3,
  nickName: '기봉이',
  introductory: '나는 기봉이'
};

export const sizeOfPicture = {
  width: '10px',
  height: '10px'
};
