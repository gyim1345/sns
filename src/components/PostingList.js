/* eslint-disable react/prop-types */
import React, { useState } from "react";
import pStore from "../stores/postingStore";
import cStore from "../stores/commentStore";
import Posting from "./Posting";

function PostingList({ postingDetail, size, user, follower }) {
  let postings =
    postingDetail === undefined
      ? pStore.postList.filter(post => post.userName === user)
      : [postingDetail];
  const { comments } = cStore;

  if (follower !== undefined) {
    const temppostings = [];
    follower.forEach(person => temppostings.push(pStore.getuserPosts(person)))
    const [temp] = temppostings;
    postings = pStore.postList.filter(post => post.userName === user)
    postings.push(...temp);
  }

  const [inputa, setInputa] = useState("");
  const [state, setState] = useState([]);

  const onChangeComment = e => {
    setInputa(e.target.value);
  };

  const addComment = (_, postId) => {
    const commentLength = cStore.commentsLength;
    cStore.createComment(postId, inputa);
    setState(commentLength);
  };

  return (
    <>
      <div>
        {postings.map(posting => (
          <ul key={posting.id}>
            <Posting
              posting={posting}
              comments={comments}
              state={state}
              setState={setState}
              addComment={addComment}
              onChangeComment={onChangeComment}
              size={size}
              user={user}
            />
          </ul>
        ))}
      </div>
    </>
  );
}

export default PostingList;
