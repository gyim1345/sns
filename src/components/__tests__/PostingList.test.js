import React from 'react';
import { mount } from 'enzyme';
import PostingList from '../PostingList';
import { post, posts, comments, currentUser, sizeOfPicture, currentUserName } from '../../Variables';
import { BrowserRouter as Router } from 'react-router-dom';

describe('<PostingList />', () => {
  it('스냅샷 비교', () => {
    const wrapper = shallow(
      <Router key="s1" keyLength={0}>
        <PostingList
          posting={posts}
          commentAPI={comments}
          currentUser={currentUser}
          userOfActivePage={currentUser}
          sizeOfPicture={sizeOfPicture}
        />
      </Router>
    );

    expect(wrapper.props().children.props.posting).toBe(posts);
    expect(wrapper.props().children.props.commentAPI).toBe(comments);
    expect(wrapper.props().children.props.currentUser).toBe(currentUser);
    expect(wrapper.props().children.props.userOfActivePage).toBe(currentUser);
    expect(wrapper.props().children.props.sizeOfPicture).toBe(sizeOfPicture);
    expect(wrapper).toMatchSnapshot();
  });
});
