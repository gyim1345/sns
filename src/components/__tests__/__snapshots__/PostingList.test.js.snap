// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostingList /> 스냅샷 비교 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <PostingList
    commentAPI={
      Array [
        Object {
          "_id": "5e90119435a2d94f9e08e35a",
          "id": 1,
          "like": Array [
            "noone@gmail.com",
          ],
          "postLId": 1,
          "replyToCommentId": null,
          "title": "title1",
          "userName": "gibong@gmail.com",
        },
      ]
    }
    currentUser="gibong@gmail.com"
    posting={
      Array [
        Object {
          "_id": "5e8ec52d9bb4913b5a844423",
          "id": 1,
          "imageUrl": "http://localhost:3000/static/images/1.jpg",
          "like": Array [
            "gibong@gmail.com",
          ],
          "tag": Array [
            "landscape",
          ],
          "title": "posting with id 1",
          "userName": "gibong@gmail.com",
        },
      ]
    }
    sizeOfPicture={
      Object {
        "height": "10px",
        "width": "10px",
      }
    }
    userOfActivePage="gibong@gmail.com"
  />
</Router>
`;
